src/main/java/christmas/service/ChristmasService.java

```diff
@@ -0,0 +1,83 @@
+package christmas.service;
+
+import christmas.constant.Constant;
+import christmas.domain.Badge;
+import christmas.domain.Benefit;
+import christmas.domain.Event;
+import christmas.domain.OrderMenu;
+import christmas.view.input.template.InputCallback;
+import christmas.view.input.template.InputTemplate;
+
+public class ChristmasService {
+
+    private final InputTemplate inputTemplate;
+
+    public ChristmasService() {
+        this.inputTemplate = new InputTemplate();
+    }
+
+    public <T> T getInputRequestResult(InputCallback<T> callback) {
+        return inputTemplate.execute(callback);
+    }
+
+    public Benefit getBenefit(OrderMenu orderMenu, int day) {
+        Benefit benefit = new Benefit();
+        addDiscountIntoBenefitDetails(orderMenu, benefit, day);
+        return benefit;
+    }
+
+    public String getOrderMenuDetails(OrderMenu orderMenu) {
+        return orderMenu.detailsToString();
+    }
+
+    public String getTotalPriceBeforeDiscount(OrderMenu orderMenu) {
+        return String.format(Constant.PRICE_UNIT + System.lineSeparator(), orderMenu.calculateTotalPrice());
+    }
+
+    public String getPresentationMenu(Benefit benefit) {
+        return benefit.presentationMenuToString();
+    }
+
+    public String getBenefitDetails(Benefit benefit) {
+        return benefit.detailsToString();
+    }
+
+    public String getTotalDiscount(Benefit benefit) {
+        return String.format(Constant.PRICE_UNIT + System.lineSeparator(), benefit.calculateTotalDiscount());
+    }
+
+    public String getTotalPriceAfterDiscount(OrderMenu orderMenu, Benefit benefit) {
+        int totalPrice = orderMenu.calculateTotalPrice();
+        int totalDiscount = benefit.calculateTotalDiscount();
+        totalPrice += (totalDiscount - benefit.calculatePresentationPrice());
+        return String.format(Constant.PRICE_UNIT + System.lineSeparator(), totalPrice);
+    }
+
+    public String getBadge(Benefit benefit) {
+        return Badge.findByTotalDiscount(benefit.calculateTotalDiscount());
+    }
+
+
+    private void addDiscountIntoBenefitDetails(OrderMenu orderMenu, Benefit benefit, int day) {
+        Event.findAllByDay(day)
+                .stream()
+                .filter(event -> orderMenu.calculateTotalPrice() > event.getCondition())
+                .forEach(event -> benefit.addDiscountIntoDetails(event, calculateDiscount(orderMenu, event, day)));
+    }
```

Comment:
benefit init을 해야 한다면, 외부에서 받아오는 것보다는 내부에서 benefit를 생성하고 반환하게 하는 게 좋아요. 외부에서 받아온 benefit에 대해 작업을 진행하면 side effect가 발생할 여지가 높아요.
해당 코드는 메소드로 분리하지 않는 게 좋았을 것 같아요.

Reply:
> benefit init을 해야 한다면, 외부에서 받아오는 것보다는 내부에서 benefit를 생성하고 반환하게 하는 게 좋아요. 외부에서 받아온 benefit에 대해 작업을 진행하면 side effect가 발생할 여지가 높아요. 해당 코드는 메소드로 분리하지 않는 게 좋았을 것 같아요.
> 
> 빌더를 활용하면 더 좋습니다 :DD

음 benefit을 반환한다는게 필드에 있는 변수를 그대로 반환해야한다는 말씀일까요??!

Reply:
외부에서 benefit을 받아오지 말고, 내부에서 생성하고 반환하는 게 좋다는 뜻이었어요.
```suggestion
    private Benefit addDiscountIntoBenefitDetails(OrderMenu orderMenu, int day) {
        Benefit benefit = new Benefit();
        Event.findAllByDay(day)
                .stream()
                .filter(event -> orderMenu.calculateTotalPrice() > event.getCondition())
                .forEach(event -> benefit.addDiscountIntoDetails(event, calculateDiscount(orderMenu, event, day)));
        return benefit;
    }
```
빌더를 만들면, 해당 작업을 빌더가 대신하게 될 거에요!

```suggestion

    private Benefit addDiscountIntoBenefitDetails(OrderMenu orderMenu, int day) {
        return new BenefitBuilder
                .orderMenu(orderMenu(
                .day(day)
                .build();
    }
```